development.markdown

```diff
@@ -137,3 +137,80 @@ More context at: https://stackoverflow.com/a/59761201/310162
 ### I get an error about `removeDirectoryRecursive`/`removeContentsRecursive`/`removePathRecursive`/`permission denied (Access is denied.)`
 
 Stack doesn't work deterministically in Windows due to mismatched expectations about how file deletion works. If you get this error, you can just retry the build and it will probably make more progress than the last time.
+
+## Building with Nix
+
+## Building package components with nix
+
+### Build the unison executable
+```
+nix build
+```
+
+### Build a specific component
+This is specified with the normal
+`<package>:<component-type>:<component-name>` triple.
+
+Some examples:
+```
+nix build '.#unison-cli:lib:unison-cli'
+nix build '.#unison-syntax:test:syntax-tests'
+nix build '.#unison-cli:exe:transcripts'
+```
+
+### Development environments
+
+#### Get into a development environment for building with stack
+This gets you into a development environment with the preferred
+versions of the compiler and other development tools. These
+include:
+
+- ghc
+- stack
+- ormolu
+- haskell-language-server
+
+```
+nix develop
+```
+
+#### Get into a development environment for building with cabal
+This gets you into a development environment with the preferred
+versions of the compiler and other development tools. Additionally,
+all non-local haskell dependencies (including profiling dependencies)
```

Comment:
non-local haskell dependencies meaning any haskell packages not included in this repo?

Reply:
that is correct